### PR TITLE
Add KCC prototype to PFI Exemplar

### PR DESCRIPTION
**Pull request by acekyd**
This pull request includes several changes to enhance the tbDEX PFI exemplar project by adding new endpoints, updating documentation, and improving the Docker setup. The most important changes include the addition of new endpoints for IDV and issuer servers, updates to the Docker configuration to expose new ports, and modifications to the README to include new example flows.

### New Endpoints and Features:

* [`src/example/idv-vendor/server.ts`](diffhunk://#diff-967317c52c2e0d55b10aa35b9ef4be14c7611df30789d807c8a2a61822039cd8R1-R43): Added a new IDV server with endpoints for handling IDV form submissions and communicating with the issuer server.
* [`src/example/issuer/server.ts`](diffhunk://#diff-303bb54959bfe1618a3b324804f1f004cc252d1a95436b386897afce64544f27R1-R218): Added a new issuer server with endpoints for handling SIOPv2 authentication requests and issuing Known Customer Credentials (KCC).

### Docker Configuration:

* [`Dockerfile`](diffhunk://#diff-8daa698cc904b421ea51420aced11c82593621ebf560091e855331e46662546dR19-R20): Exposed ports 3001 and 3002 for the new issuer and IDV servers.
* [`docker-compose.yaml`](diffhunk://#diff-65c4a395ab0cf46bdb9cec0456962a490ca2750426b37abf8aebe79efc527a31R22-R23): Updated to expose ports 3001 and 3002 for the new services.

### Documentation Updates:

* [`README.md`](diffhunk://#diff-39a35395c72a5edfc8dcc8d74172d4c64f086a6af5a90c61eff935d46d899ec1L5-R19): Updated to include new example flows for issuing Known Customer Credentials (KCC) and running the mock Identity Vendor (IDV) and Issuer servers. [[1]](diffhunk://#diff-39a35395c72a5edfc8dcc8d74172d4c64f086a6af5a90c61eff935d46d899ec1L5-R19) [[2]](diffhunk://#diff-39a35395c72a5edfc8dcc8d74172d4c64f086a6af5a90c61eff935d46d899ec1R135-R143)

### Script and Configuration Updates:

* [`container-start.sh`](diffhunk://#diff-fac849758aaf19515f030ad073d83eeaad744f67642107c9641f527522cef496R3-R17): Added logging and commands to run the example issuer creation and seed offerings scripts.
* [`package.json`](diffhunk://#diff-ca603e3eeb47cce631b27ec8af6a61ce0782c36e491e68d569723e6e6e0e4c1bR58): Added new npm script `example-wallet-issue-kcc` for running the full flow of issuing a KCC.

### Example Flows:

* Added a series of scripts in `src/example/kcc-wallet/` to demonstrate the full flow of creating a DID, resolving IDV service endpoints, submitting SIOPv2 responses, and requesting credentials. [[1]](diffhunk://#diff-855d830d929ec472e12ee43299329ccfdec0db964165ea7c1445d04a6a56013fR1-R7) [[2]](diffhunk://#diff-472b979b45bfabb61e3ed72660b75d98343a61dda6d039d593303df6d5b957b5R1-R12) [[3]](diffhunk://#diff-839173b05bd0f5801bab305cbd725e48aa08910021b4d5786212333b180b791eR1-R6) [[4]](diffhunk://#diff-5fccc4c945d06bdf15725b8f56e11a6c3fe6f42e7f06d67f8597e1ee765e8317R1-R29) [[5]](diffhunk://#diff-71386e309f659efb210dc15b813b949665822dab05bbfc81f2330e49872e0787R1-R19) [[6]](diffhunk://#diff-80c7284a66deeb20d1ac2b31cccb93d03ebb1dafbfb83ee68baa5a55857a1debR1-R6) [[7]](diffhunk://#diff-046549baa40caf3421f6b35b8a166d2f0700622ee77238f8b1531b6fb3cea02cR1-R6) [[8]](diffhunk://#diff-74ede719b064978ef6b152f524b458d0d77f76fd35eace52d55ca005a123b55cR1-R21) [[9]](diffhunk://#diff-55d5ba4715dc6b88605e4bd5c946581b1d75d404e1f37b57509a34091b693ecbR1-R38) [[10]](diffhunk://#diff-eaa7216b3a3f01352fc411eb60469e36e1a221c029aa6d74ea9db238a9fc9f4dR1-R41)

These changes significantly enhance the functionality and documentation of the tbDEX PFI exemplar project, making it easier to follow and implement the full flow of issuing Known Customer Credentials.

### CODE REVIEW

**Comment by finn-tbd** [comment]
Those kcc wallet scripts look a lot like the ones that already exist at `javascript/kcc-prototype-exemplar/wallet` - are we sure we want to duplicate them?

**Comment by acekyd** [comment]
> Those kcc wallet scripts look a lot like the ones that already exist at `javascript/kcc-prototype-exemplar/wallet` - are we sure we want to duplicate them?

You're correct. They are an integration of that prototype into the existing PFI exemplar to execute the KCC along with the offerings as a way to show end to end tbDEX interop.

File: javascript/tbdex-pfi-exemplar/src/seed-offerings.ts [hunk 1]
```diff
@@ -224,7 +224,7 @@ async function main() {
                     path: ['$.type[*]'],
                     filter: {
                       type: 'string',
-                      pattern: '^SanctionCredential$',
+                      pattern: '^KnownCustomerCredential$',
```
Comment (angiejones) [inline javascript/tbdex-pfi-exemplar/src/seed-offerings.ts:227]: there was a change to the spec. it no longer looks for `type`. It looks for `credentialSchema.id`:

```json
              {
                "path": [
                  "$.credentialSchema[*].id"
                ],
                "filter": {
                  "type": "string",
                  "const": "https://vc.schemas.host/kcc.schema.json"
                }
              }
```

**Review by angiejones — CHANGES_REQUESTED** [review]

**Comment by michaelneale** [comment]
I like this - I think this could be the right home for end to end tbdex including with kcc as it is important. I guess open question if that external example kcc is retained, or added in parallel to the pfi-exemplar for just that bit? I think it makes more sense in context of this so I think this is EXCELLENT!

Nice one @acekyd

**Comment by michaelneale** [comment]
not sure if missing it but should the kcc be used as part of the end to end tbdex flow as well?

File: javascript/tbdex-pfi-exemplar/src/seed-offerings.ts (hunk 1 above)
Comment (acekyd) [reply javascript/tbdex-pfi-exemplar/src/seed-offerings.ts:227]: Oh gotcha. Thanks for catching that.

**Review by finn-tbd — APPROVED** [review]

**Review by angiejones — APPROVED** [review]

**Pull request merged**